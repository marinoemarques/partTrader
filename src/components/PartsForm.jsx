import React, {useState} from "react";
import {useExclusions, useFetched} from "../lookups/APIs";

export function PartsForm() {
    const [partID, setPartID] = useState("");
    const [message, setMessage] = useState("");
    const exclusions = useExclusions();
    const fetched = useFetched();

    const listsCheck = (partID) =>
        !partID.match(/^(\d{4})(-)([a-zA-Z0-9]){4,}$/) ?
            setMessage('please check your part number format') :
        exclusions.includes(partID) ?
            setMessage('this part is excluded') :
        fetched.includes(partID) ?
            setMessage(`we found ${partID} in our system`) :
            setMessage('no parts found')

    return (
        <form onSubmit={(e) => {
            e.preventDefault();
            listsCheck(partID);
        }}>
            <label>
                Which part number you are looking for?
                <input type="text" placeholder="1234-example" value={partID} onChange={e => setPartID(e.target.value)} />
            </label>
            <input type="submit" value="Submit"/>
            <div>
                {message}
            </div>
        </form>
    );
}