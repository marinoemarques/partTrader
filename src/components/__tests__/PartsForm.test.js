import React from 'react';
import { render } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import PartsForm from "../PartsForm";

describe('PartsForm', () => {
    it('renders the lookup form', async () => {
        const { getByText } = render(
            <PartsForm />
        );
        expect(getByText('Which part number are you looking for?')).toBeInTheDocument();
    });
});
