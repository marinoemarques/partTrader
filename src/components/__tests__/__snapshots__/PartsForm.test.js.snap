// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PartsForm should match snapshot 1`] = `
<form>
  <label>
    Which part number are you looking for?
    <div>
      <input
        class="sc-bdfBQB ea-DHut"
        placeholder="1234-example"
        type="text"
        value=""
      />
      <input
        class="sc-bdfBQB ea-DHut"
        type="submit"
        value="Submit"
      />
    </div>
  </label>
  <div
    class="sc-gsTEea gdzeFY"
  />
</form>
`;
